### PR TITLE
[PE1-1505] CDN Class "kind" implementation

### PR DESCRIPTION
**Pull request by caiofralmeida**
## Description
The idea is to avoid configuring types of "values-*.yaml" to provide different configurations and have a Kubernetes kind giving the hosted zone and certificate for each "class" configured. In this way, the developer should provide only the "class" desired without concern about hosted zones and certificate configurations.

```yaml
apiVersion: cdn.gympass.com/v1alpha1
kind: CDNClass
metadata:
  name: foo-com
spec:
  certificateArn: "<Certificate ARN from given hosted zone>"
  hostedZoneID: "<foo-com hosted zone ID>"
```

By the way, this kind may have more parameters in the future, and we can refactor some configurations from the App config as well.

## Motivation and Context
- CDN Ingress interface https://gympass.atlassian.net/browse/PE1-1505

## How has this been tested?
Tested manually via local environment (I'm going to test through pe-develop Kubernetes cluster as well)

## Checklist

- [x] I have implemented automated tests for the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/k8s/cdnclass.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+package k8s
+
+// CDNClass ...
```
Comment (monicajss) [inline internal/k8s/cdnclass.go:22]: I think you need to continue the text... Right?

File: internal/k8s/cdnclass.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+type CDNClass struct {
+	// CertificateArn ...
```
Comment (monicajss) [inline internal/k8s/cdnclass.go:24]: same here

File: internal/k8s/cdnclass.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	CertificateArn string
+	// HostedZoneID ...
```
Comment (monicajss) [inline internal/k8s/cdnclass.go:26]: and here

**Review by monicajss — CHANGES_REQUESTED** [review]

**Review by go-faustino — APPROVED** [review]
Great work, this is really useful!!!

Left a couple of comments, which can be addressed later, but overall LGTM!

File: api/v1alpha1/cdnclass_types.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+	// +kubebuilder:validation:Required
+	CertificateArn string `json:"certificateArn"`
+	// +kubebuilder:validation:Required
+	HostedZoneID string `json:"hostedZoneID"`
+}
+
+// CDNClassStatus defines the observed state of CDNClass
+type CDNClassStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
```
[the remark this answers is not in the document]
Comment (caiofralmeida) [reply api/v1alpha1/cdnclass_types.go:42]: There is no controller to manage this resource, until then it's just a "store" of the settings, I couldn't think of what information to put in the status unless I create a controller for that, I don't know if it's worth it at this point. What do you think?

[the remark this answers is not in the document]
Comment (caiofralmeida) [reply api/v1alpha1/cdnclass_types.go:42]: Anyway, I configured the field Conditions to the status:
```go
// CDNClassStatus defines the observed state of CDNClass
type CDNClassStatus struct {
	// CDNClass resource condition
	// +optional
	// +nullable
	Conditions []metav1.Condition `json:"conditions"`
}
```

**Pull request merged**